lib/pbench/server/api/resources/query_apis/__init__.py

```diff
@@ -399,83 +383,97 @@ def _call(
+        except Exception as e:
+            self.logger.exception(
+                "Unexpected problem postprocessing Elasticsearch response {}: {}",
+                es_response.json(),
+                e,
+            )
+            abort(500, message="INTERNAL ERROR")
 
     def post(self):
         """
-        Handle a POST operation involving a call to the server's Elasticsearch
-        instance. The assembly and post-processing of the Elasticsearch query
-        are handled by the subclasses through the assemble() and postprocess()
-        methods; we do basic parameter validation and conversions here along
-        with handling Elasticsearch errors in a single place.
+        Handle a Pbench server POST operation that will involve a call to the
+        server's configured Elasticsearch instance. The assembly and
+        post-processing of the Elasticsearch query are handled by the
+        subclasses through the assemble() and postprocess() methods;
+        we do basic parameter validation and conversions here.
+
+        Each subclass defined a set of supported JSON parameters: this
+        code processes each JSON parameter in the dict to perform type
+        validation/conversion where necessary. Missing "required" parameters
+        and any parameter with a null value are rejected. Parameters that
+        are not in the class schema are ignored but logged.
+
+        TODO: We have not completely resolved the API accommodations for
+        queries without an authenticated user. I'm tending towards the idea
+        of an "access": "public" or "published": True parameter; however we
+        might decide to allow "owner": None or omitting the "owner" parameter
+        entirely, in which case this logic will need to be adjusted.
         """
         json_data = request.get_json(silent=True)
         if not json_data:
             self.logger.warning("Invalid JSON object. Query: {}", request.url)
             abort(400, message="Invalid request payload")
 
         missing_keys = [
-            k.name
-            for k in self.schema
-            if (k.name not in json_data and k.required)
-            or (k.name in json_data and not json_data[k.name])
+            n
+            for n, p in self.schema
+            if (n not in json_data and p.required)
+            or (n in json_data and not json_data[n])
         ]
         if len(missing_keys) > 0:
             self.logger.warning("Required keys {} are missing from JSON", missing_keys)
             abort(400, message=f"Missing required parameters: {','.join(missing_keys)}")
 
-        string_data = json_data.copy()
         for p in json_data:
             tp = self.schema.get(p)
             if tp:
                 original = json_data[p]
                 try:
                     value = tp.type.convert(original)
+                except ConversionError as e:
+                    self.logger.warning("{}", str(e))
```

Comment:
Won't this work if you omit the first parameter?

Reply:
No, actually, it won't, and it drove me insane trying to figure this out yesterday. There's a unit test case that passes a text "JSON" body and the logger explodes trying to interpret "`{'babble': 42}"` as a formatting directive.

Reply:
LOL!  I think this deserves a code comment...so that we don't come along later and say, "Hey, that's pointless, let me just take that out...."  :-D